app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,27 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+    @params = params
+  end
+
+  def create
+    @post = Post.new(posts_params)
+    if @post.save
+      redirect_to user_post_path(id: @post.id, user_id: @post.user_id), notice: 'Post created succesfully!'
+    else
+      render :new, alert: 'Post could not be created an Error occurred!'
+    end
+  end
+
+  private
+
+  def posts_params
+    params.require(:post).permit(:title, :text, :user_id)
   end
```

Comment:
- [OPTIONAL] You can also use `@current_user` to build content for your resource(posts, likes, comments, etc) because its available across all the controllers by inheritance from `ApplicationController`, that way your code is easier to read 😄